internal/request/request.go

```diff
@@ -0,0 +1,200 @@
+package request
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/RedHatInsights/catalog_mqtt_client/internal/catalogtask"
+	"github.com/RedHatInsights/catalog_mqtt_client/internal/common"
+	"github.com/RedHatInsights/catalog_mqtt_client/internal/jsonwriter"
+	"github.com/RedHatInsights/catalog_mqtt_client/internal/logger"
+	"github.com/RedHatInsights/catalog_mqtt_client/internal/tarwriter"
+	"github.com/RedHatInsights/catalog_mqtt_client/internal/towerapiworker"
+	mqtt "github.com/eclipse/paho.mqtt.golang"
+	log "github.com/sirupsen/logrus"
+)
+
+type PageWriter interface {
+	Write(name string, b []byte) error
+	Flush() error
+	FlushErrors(msg []string) error
+}
+
+// RequestHandler interface allows for easy mocking during testing
+type RequestHandler interface {
+	StartHandlingRequests(mqttClient mqtt.Client, config *common.CatalogConfig, wh towerapiworker.WorkHandler)
```

Comment:
@bzwei We need to pass in the FactoryInterface for CatalogTask in here and sub it out during testing.